### PR TITLE
misc(core): Add new shard key _ws_ and rename _ns to _ns_

### PR DESCRIPTION
**Pull request by tjackpaul**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [x] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :**
We have `_ns, __name__` as shard key.

**New behavior :**

1. All internal reserved label names to start and end with single underscore(`_`). Hence, the shard key `_ns` will be updated to `_ns_`. 
2. Also, adding workspace `_ws_` to the shard key.

**BREAKING CHANGES**

Users will not be able to query Timeseries with old shard key. Hence, the idea is to setup and run a new cluster with new shard key changes in parallel to the existing cluster until the TTL is met.

### CODE REVIEW

File: core/src/main/resources/filodb-defaults.conf [hunk 1]
```diff
@@ -24,17 +24,17 @@ filodb {
     columns = ["metric:string", "tags:map"]
 
     # Predefined keys allow space to be saved for over the wire tags with the given keys
-    predefined-keys = ["_ns", "app", "__name__", "instance", "dc", "le", "job", "exporter"]
+    predefined-keys = ["_ws_", "_ns_", "app", "__name__", "instance", "dc", "le", "job", "exporter"]
```
Comment (velvia) [inline core/src/main/resources/filodb-defaults.conf]: NOTE: changing predefined keys will require a clean wipe of all data.
Could you please add a warning comment above that one should only add to predefined keys.  Changing existing keys will cause incorrect tags to be reported and probably require a clean wipe, thanks.

File: doc/downsampling.md [hunk 2]
```diff
@@ -50,7 +50,7 @@ Downsampling for prometheus counters will come soon.
 Downsampling is configured at the time of dataset creation. For example:
 
 ```
-./filo-cli -Dconfig.file=conf/timeseries-filodb-server.conf  --command create --dataset prometheus --dataColumns timestamp:ts,value:double --partitionColumns tags:map --shardKeyColumns __name__,_ns --downsamplers "tTime(0),dMin(1),dMax(1),dSum(1),dCount(1),dAvg(1)"
+./filo-cli -Dconfig.file=conf/timeseries-filodb-server.conf  --command create --dataset prometheus --dataColumns timestamp:ts,value:double --partitionColumns tags:map --shardKeyColumns __name__,_ns_,_ws_ --downsamplers "tTime(0),dMin(1),dMax(1),dSum(1),dCount(1),dAvg(1)"
```
Comment (velvia) [inline doc/downsampling.md]: Oh, please remove all examples involving `create`, this is no longer a valid CLI command (configs are used only).

File: core/src/main/resources/filodb-defaults.conf (hunk 1 above)
Comment (tjackpaul) [reply core/src/main/resources/filodb-defaults.conf]: This is taken care of.

**Comment by tjackpaul** [comment]
@velvia Updated the readme and other downsampling doc

File: README.md [hunk 3]
```diff
@@ -385,7 +385,7 @@ The number of shards in each dataset is preconfigured in the source config.  Ple
 
 Metrics are routed to shards based on factors:
 
-1. Shard keys, which can be for example an application and the metric name, which define a group of shards to use for that application.  This allows limiting queries to a subset of shards for lower latency.
+1. Shard keys, which can be for example an application and the metric name, which define a group of shards to use for that application.  This allows limiting queries to a subset of shards for lower latency. Currently `_ws_`, `_ns_`, `__name__` labels are mandatory to calculate shard key.
```
Comment (velvia) [inline README.md]: Not quite right - it's going to be the `metric` column of partition key, not the `__name__` label

**Pull request merged**